src/main/scala/ai/privado/entrypoint/ScanProcessor.scala

```diff
@@ -260,7 +260,15 @@ object ScanProcessor extends CommandProcessor {
     AppCache.init(sourceRepoLocation)
     import io.joern.console.cpgcreation.guessLanguage
     println("Guessing source code language...")
-    val xtocpg = guessLanguage(sourceRepoLocation) match {
+    var languageDetected: Option[String] = None
+    try {
+      languageDetected = guessLanguage(sourceRepoLocation)
+    } catch {
+      case exc: Exception =>
+        println(s"Error Occurred: ${exc.toString}")
+        exit(1)
+    }
+    val xtocpg = languageDetected match {
       case Some(lang) =>
```

Comment:
@abhstabs A better way to write this can be, what is your thought?

```
val xtocpg  = Try(guessLanguage(sourceRepoLocation)) match {
 case Success(langDetected) => 
     langDetected match {
         case Some(lang) => ....
         case _ => .....
     }
    case Failure(exc) => 
    println(s"Error Occurred: ${exc.toString}")
       exit(1)

}
```